[PGO] GPUP's executable should not be built for instrumentation

emw-apple opened the pull request:
#### f35c6c8e306f867e27bdcbf6c2452916575c99dd
<pre>
[PGO] GPUP&apos;s executable should not be built for instrumentation
<a href="https://bugs.webkit.org/show_bug.cgi?id=261709">https://bugs.webkit.org/show_bug.cgi?id=261709</a>
rdar://115688644

Reviewed by Alexey Proskuryakov.

Like other XPC services, its use of the WebKit *frameworks* should be
instrumented, but not its main executable. Copy the
CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING = NO build setting from
other XPC service targets.

* Source/WebKit/Configurations/GPUService.xcconfig:

Canonical link: <a href="https://commits.webkit.org/268164@main">https://commits.webkit.org/268164@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3aae8c14d442c1866dc61eaece09cd46a1702d71

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18705 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19046 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19649 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20569 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17511 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22355 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19187 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19347 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18930 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19079 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16284 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21448 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16302 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23498 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17327 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17216 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21392 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17820 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15124 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16879 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4490 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21246 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17659 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/Configurations/GPUService.xcconfig:
```diff
@@ -28,6 +28,7 @@ PRODUCT_BUNDLE_IDENTIFIER = $(PRODUCT_NAME);
 INFOPLIST_FILE[sdk=embedded*] = GPUProcess/EntryPoint/Cocoa/XPCService/GPUService/Info-iOS.plist;
 INFOPLIST_FILE[sdk=macosx*] = GPUProcess/EntryPoint/Cocoa/XPCService/GPUService/Info-OSX.plist;
 
+CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING = NO; // Disable PGO profile generation
```
Looks like something went wrong with this PR, this empty line is all I see.

aproskuryakov replied:
Must have been a GitHub glitch.

emw-apple committed:
[PGO] GPUP's executable should not be built for instrumentation
https://bugs.webkit.org/show_bug.cgi?id=261709
rdar://115688644

Reviewed by Alexey Proskuryakov.

Like other XPC services, its use of the WebKit *frameworks* should be
instrumented, but not its main executable. Copy the
CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING = NO build setting from
other XPC service targets.

* Source/WebKit/Configurations/GPUService.xcconfig:

Canonical link: https://commits.webkit.org/268164@main
